src/services/housing.js

```diff
@@ -171,10 +179,28 @@ const saveApartmentApplication = async (applicationDetails) => {
   };
 
   const applicationID = applicationDetails.ApplicationID;
-  if (applicationID > 0) {
-    return await http.put(`housing/apartment/applications/${applicationID}/`, applicationDetails);
-  } else {
-    return await http.post(`housing/apartment/applications/`, applicationDetails);
+  let result = null;
+  try {
+    if (applicationID > 0) {
+      result = await http.put(
+        `housing/apartment/applications/${applicationID}/`,
+        applicationDetails,
+      );
+    } else {
+      result = await http.post(`housing/apartment/applications/`, applicationDetails);
+    }
+  } catch (err) {
+    if (err?.status === 401 || err?.name?.includes('AuthError')) {
+      console.log('Received 401 (Unauthorized)');
+    } else if (err?.status === 404 || err?.name?.includes('NotFound')) {
+      console.log(
+        'Received 404 indicates that the requested application was not found in the database',
+      );
+    } else {
+      throw err;
+    }
+  } finally {
+    return result;
```

Comment:
Many of these endpoints need to use the exact same structure of 
``` javascript
let result = null
try {
   result = await DoSomething
} catch (err) {
  if (it's a 401 error) {
    console.log('Message that 401')
  } else if (it's a 404 error) {
    console.log('Custom message about meaning of 404)
  } else {
    throw err
  }
} finally {
  return result;
}
```

Since much of that `try ... catch ... finally` needs to be identical across multiple functions in housing.js, is there a way to put that in a function/wrapper to reduce the repetitive lines of code?

The tricky part is somehow changing the API request and log text while leaving the rest the same (the text is trival, just pass a string. It is the API requests that I am not sure about, since we need `http.get`, `http.put`, `http.post`, etc.)

Reply:
I suspect callbacks might be a solution

Reply:
Improvements coming in #1100